### ISSUE TITLE
2.2.1 cypher failure / combination  optional match  and path-length

### ISSUE THREAD

**Issue by tbaum** [neo4j/neo4j#4479]
since 2.2.1 a statement like

```
MATCH (root:Root) 
OPTIONAL MATCH (root)-[:REL]-(rel1) 
OPTIONAL MATCH (rel1)-[:REL*0..1]-(rel2) 
RETURN rel2
```

failes `Expected to find a node at rel1 but found null instead`

**Comment by lutovich** [comment]
Hello @tbaum, I believe this issue was fixed by https://github.com/neo4j/neo4j/pull/4454.
Could you please try 2.3.0-M01? It should include the fix.

**Comment by tbaum** [comment]
Hej Konstantin,  

yes it is fixed in 2.3.0-M01 thank you! 

will it go into 2.2.x?

cheers, Thomas

**Comment by lutovich** [comment]
@tbaum yes, it will be released with 2.2.2. Thanks for report!
